src/sql/workbench/test/electron-browser/api/extHostModelView.test.ts

```diff
@@ -44,335 +45,444 @@ suite('ExtHostModelView Validation Tests', () => {
+		});
 
-	test('Form component groups are handled correctly by adding each item in the group and a label to the form', () => {
-		// Set up the mock proxy to save the component that gets initialized so that it can be verified
-		let rootComponent: IComponentShape;
-		mockProxy.setup(x => x.$initializeModel(It.isAny(), It.isAny())).callback((handle, componentShape) => rootComponent = componentShape);
-		mockProxy.setup(x => x.$addToContainer(It.isAny(), It.isAny(), It.isAny(), undefined)).returns(() => Promise.resolve());
+		test('Inserting component give negative number fails', () => {
+			mockProxy.setup(x => x.$initializeModel(It.isAny(), It.isAny())).callback((handle, componentShape) => { });
+			mockProxy.setup(x => x.$addToContainer(It.isAny(), It.isAny(), It.isAny(), undefined)).returns(() => Promise.resolve());
+			mockProxy.setup(x => x.$addToContainer(It.isAny(), It.isAny(), It.isAny(), It.isAny())).returns(() => Promise.resolve());
+			mockProxy.setup(x => x.$removeFromContainer(It.isAny(), It.isAny(), It.isAny())).returns(() => Promise.resolve());
 
-		// Set up the form with a top level component and a group
-		let topLevelList = modelView.modelBuilder.listBox().component();
-		let groupInput = modelView.modelBuilder.inputBox().component();
-		let groupDropdown = modelView.modelBuilder.dropDown().component();
+			// Set up the form with a top level component and a group
+			let listBox = modelView.modelBuilder.listBox().component();
+			let inputBox = modelView.modelBuilder.inputBox().component();
+			let dropDown = modelView.modelBuilder.dropDown().component();
 
-		let topLevelInputFormComponent: azdata.FormComponent = { component: topLevelList, title: 'top_level_input' };
-		let groupInputFormComponent: azdata.FormComponent = { component: groupInput, title: 'group_input' };
-		let groupDropdownFormComponent: azdata.FormComponent = { component: groupDropdown, title: 'group_dropdown' };
+			let flex = modelView.modelBuilder.flexContainer().withItems([listBox, inputBox]).component();
+			modelView.initializeModel(flex);
 
-		let groupTitle = 'group_title';
+			const itemConfigs: InternalItemConfig[] = (flex as IWithItemConfig).itemConfigs;
+			assert.equal(itemConfigs.length, 2, `Unexpected number of items in list. Expected 2, got ${itemConfigs.length} ${JSON.stringify(itemConfigs)}`);
+			assert.throws(() => flex.insertItem(dropDown, -1), `Didn't get expected exception when calling insertItem with invalid index -1`);
+		});
 
-		// Give the group a default layout and add one just for the input component too
-		let defaultLayout: azdata.FormItemLayout = {
-			horizontal: true
-		};
-		let groupInputLayout: azdata.FormItemLayout = {
-			horizontal: false
-		};
+		test('Inserting component give wrong number fails', () => {
```

Comment:
do we have any tests around ensuring that we can't insert an item with an id that already exists?

Reply:
There's currently nothing preventing that from being done AFAIK. I'd say the behavior for that would be undefined at best currently. At some point we may want to define better behavior around that but given that we haven't seen any issues with it so far I'm going to leave that until we have some reason to need/want to do that extra work.